modules/loader/src/main/java/com/opengamma/strata/loader/csv/LegalEntityRatesCurvesCsvLoader.java

```diff
@@ -207,24 +207,28 @@
       for (Map.Entry<CurveGroupName, Map<Pair<RepoGroup, Currency>, CurveName>> repoEntry : repoGroups.entrySet()) {
         CurveGroupName groupName = repoEntry.getKey();
         Map<Pair<RepoGroup, Currency>, Curve> repoCurves = MapStream.of(repoEntry.getValue())
-            .mapValues(name -> curves.get(name))
-            .peek(e -> {
-              if (e.getValue() == null) {
-                throw new IllegalArgumentException(
-                    "Repo curve values for " + e.getKey().toString() + " in group " + groupName.getName() +
-                        " are missing on " + date.toString());
-              }
-            })
+            .mapValues(
+                name -> {
+                  Curve curve = curves.get(name);
```

Comment:
We try to avoid blocks by creating extra methods. So, this would be `name -> queryCurve(name, curves, "Repo")

(this extra method can be used for both of these cases)